account-decoder/src/parse_config.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+    parse_account_data::{ParsableAccount, ParseAccountError},
+    validator_info,
+};
+use bincode::deserialize;
+use serde_json::Value;
+use solana_config_program::{get_config_data, ConfigKeys};
+use solana_sdk::pubkey::Pubkey;
+use solana_stake_program::config::Config as StakeConfig;
+
+pub fn parse_config(data: &[u8], pubkey: &Pubkey) -> Result<ConfigAccountType, ParseAccountError> {
+    let parsed_account = if pubkey == &solana_stake_program::config::id() {
+        get_config_data(data)
+            .ok()
+            .and_then(|data| deserialize(data).ok())
+            .map(ConfigAccountType::StakeConfig)
+    } else {
+        deserialize::<ConfigKeys>(data).ok().and_then(|key_list| {
+            if !key_list.keys.is_empty() && key_list.keys[0].0 == validator_info::id() {
+                parse_config_data::<String>(data, key_list.keys).and_then(|validator_info| {
```

Comment:
Since you have already deserialized the key list, what if you called `parse_config_data` with only the config data bytes?

Reply:
Could do that. The keys need to be stringified as well, though. My thinking was to do that all in one place, as there may be other live config account types in the future (SignedUpdateManifest, eg) that will need all that treatment. Could rename the method, if `parse_config_data` is too specific.

Reply:
I don't feel too strongly about this, feel free to skip